autopts/utils.py

```diff
@@ -477,6 +490,70 @@ def log_memory_usage():
     logging.debug(f"Memory usage: {mem_usage:.2f} MB")
 
 
+def extract_wid_testcases_to_csv():
+
+    # Example log block format:
+    # BEGIN OnImplicitSend:
+    # wid: 35
+    # test_case_name: GAP/ADV/BV-01-C
+    # ...
+    # END OnImplicitSend
+
+    profile_wid_map = defaultdict(lambda: defaultdict(set))
+
+    for log_file in LOG_DIR.rglob("*.log"):
+        logging.debug(f"Processing log file: {log_file}")
+        try:
+            with log_file.open(encoding='utf-8', errors='ignore') as f:
+                lines = f.readlines()
+        except Exception as e:
+            logging.warning(f"Failed to read {log_file}: {e}")
+            continue
+
+        # Track wheter inside an OnImplicitSend block
+        in_implicit_send_block = False
+        wid = None
+        test_case_name = None
+
+        for line in lines:
+            stripped = line.strip()
+
+            # Detect start of a new block
+            if not in_implicit_send_block and stripped.startswith("BEGIN OnImplicitSend:"):
+                in_implicit_send_block = True
+                wid = None
+                test_case_name = None
+
+            elif in_implicit_send_block:
```

Comment:
Up to you, but it should be possible to replace the loop with a regex. 

Something like 

```python
pattern = r"""
BEGIN OnImplicitSend:
.*?
wid:\s*(?P<wid>\d+)
.*?
test_case_name:\s*(?P<test_case_name>\w+/\w+/\w+-\w+)
.*?
END OnImplicitSend:
"""

result = re.search(pattern, text)

if result:
    wid = result["wid"]
    test_case_name = result["test_case_name"]
    print(f"wid: {wid}")
    print(f"test_case_name: {test_case_name}")
```

If you prefer the loop, that's also fine